user-console/src/main/resources-filtered/org/pentaho/mantle/public/browser/index.jsp

```diff
@@ -269,7 +269,20 @@
         else if (paramJson.eventSubType == "RefreshBrowsePerspectiveEvent") {
           //Clear the Browse Perspective cache
           window.parent.mantle_isBrowseRepoDirty = true;
-          FileBrowser.update(window.parent.HOME_FOLDER, paramJson.booleanParam); // refresh folder list
+          var path;
+          var folderClicked = FileBrowser.fileBrowserModel.getFolderClicked();
```

Comment:
I believe this logic is already being done in `FileBrowser.update`. Please check, possibly fix a root bug, if there is one.